### ISSUE TITLE
Failed to Create Network Controller Service

### ISSUE THREAD

**Issue by krisjturn** [Microsoft/SDN#149]
We are t tying to deploy network service to a newly deployed environment using VMM.  We get as far as deploying applications to the 3 vm's.  Each time we try to deploy the service we get 2 of the 3 vm's to get to Invoke script command Pre-Install_5 for Application profile Completed.  The 3rd VM never successfully runs the Pre-Install_5 job.  So after some time the other 2 time out and fail then the service deployment job fails.  I have checked the logs on the NC and can't find anything in the err logs to show why the 3rd VM never gets past that point.  

We have an open ticket with Premier support but haven't really moved forward with any solutions yet.

**Comment by VitCJr** [comment]
two things: 
1. If you're deploying NC via VMM - make sure you selected proper scenario (e.g.: domain based certificates or self-signed) - it depence on your environment, and all pre-requests were configured correctly. Check that VMs joined domain correctly, check if there's all required root certificates on NC nodes. We're using domain certs from our CA and we distributed ROOT CA cert by group policy - we noticed that sometimes  GPO "freezes" for a while and NC node doesn't get CA cert. That can potentially lead to this problem, just fix certificate issue and restart job (if you see some group policy related errors in VMM Job); 
2. Check DNS records (are they in place, are they correct? If you tried to install NC multiple times your VMM can assign each time new IP from the pool. In large environment issue will be related to DNS replication things.




FYI
Also,the issue that you'll face later: is multi-homed VMs of SLB and gateways. On this VMs should be only one default gateway configured. And if you have complex environment, you will need to do some custom persistence routes and remove default GW from internal adapters. If you won't do it - you will face same issue- some of the Apps won't be installed.
